File: test_feature/companies/consulting/src/main/java/com/github/klee0kai/stone/test_feature/consulting/di/ConsultingComponent.java
package com.github.klee0kai.stone.test_feature.consulting.di;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.annotations.module.BindInstance;
import com.github.klee0kai.stone.test_feature.finance.di.AccountingModule;
import com.github.klee0kai.stone.test_feature.finance.model.WorkCalendar;
import com.github.klee0kai.stone.test_feature.hr.di.DepartmentsModule;
import com.github.klee0kai.stone.test_feature.planning.di.ProjectsModule;

@Component
public interface ConsultingComponent extends ConsultingDependencies {

    DepartmentsModule departmentsModule();

    AccountingModule accountingModule();

    ProjectsModule projectsModule();

    @BindInstance(cache = BindInstance.CacheType.Strong)
    WorkCalendar workCalendar(WorkCalendar workCalendar);


}

File: test_feature/companies/consulting/src/main/java/com/github/klee0kai/stone/test_feature/consulting/di/ConsultingComponent.kt
package com.github.klee0kai.stone.test_feature.consulting.di

import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.stone.annotations.module.BindInstance
import com.github.klee0kai.stone.test_feature.finance.di.AccountingModule
import com.github.klee0kai.stone.test_feature.finance.model.WorkCalendar
import com.github.klee0kai.stone.test_feature.hr.di.DepartmentsModule
import com.github.klee0kai.stone.test_feature.planning.di.ProjectsModule

@Component
interface ConsultingComponent : ConsultingDependencies {

    fun departmentsModule(): DepartmentsModule?

    fun accountingModule(): AccountingModule?

    fun projectsModule(): ProjectsModule?

    @BindInstance(cache = BindInstance.CacheType.Strong)
    fun workCalendar(workCalendar: WorkCalendar?): WorkCalendar?
}
